homeassistant/util/__init__.py

```diff
@@ -44,7 +44,7 @@ def slugify(text: str | None, *, separator: str = "_") -> str:
     """Slugify a given text."""
     if text == "" or text is None:
         return ""
-    slug = unicode_slug.slugify(text, separator=separator)
+    slug = unicode_slug.slugify(str(text), separator=separator)
```

Comment:
Our function `slugify` only expects a string or `None`. If anything else is passed the function is used incorrectly. We shouldn't copy to string here.

Reply:
Indeed but there is no test for such a case and [it _is_ possible to pass non-str values](https://github.com/home-assistant/core/pull/108753#discussion_r1466299648).

Anyway, are you saying that we should check on the integration side before passing through a non-str? If so, I would appreciate your [peer review](https://github.com/home-assistant/core/pull/108937) if you had a moment on the same.

Reply:
Normally you know what type you are handling and don't need to check.